edit-post/api/components/plugin-sidebar/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Children, cloneElement, compose } from '@wordpress/element';
+import { Slot, Fill, withFocusReturn } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { withPluginContext } from '../../plugin';
+import SidebarLayout from './sidebar-layout';
+import ErrorBoundary from './error-boundary';
+
+/**
+ * Name of slot in which the sidebar should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginSidebar';
+
+/**
+ * The plugin sidebar fill.
+ */
+class PluginSidebar extends Component {
+	constructor( props ) {
+		super( props );
+
+		if ( typeof props.name !== 'string' ) {
+			throw 'Sidebar names must be strings.';
+		}
+		if ( ! /^[a-z][a-z0-9-]*$/.test( props.name ) ) {
+			throw 'Sidebar names must include only lowercase alphanumeric characters or dashes, and start with a letter. Example: "my-sidebar".';
+		}
+	}
+
+	render() {
+		if ( ! this.namespacedName ) {
+			this.namespacedName = `${ this.props.pluginContext.namespace }/${ this.props.name }`;
+		}
+		if ( this.props.activePlugin !== this.namespacedName ) {
+			return null;
+		}
+
+		const { children, ...props } = this.props;
+		const newProps = {
+			...props,
+			namespacedName: this.namespacedName,
+		};
+
+		return (
+			<Fill name={ SLOT_NAME }>
+				<SidebarLayout
+					title={ props.title }
+					onClose={ props.onClose } >
+					<ErrorBoundary pluginName={ this.namespacedName }>
+						{ cloneElement( Children.only( children ), newProps ) }
```

Comment:
`Children.only` is not accommodating to different types of elements. This will error when rendering a string:

```js
const { createElement } = wp.element;
const { registerPlugin, PluginSidebar } = wp.editPost.__experimental;

function MyPlugin() {
	return createElement(
		PluginSidebar,
		{ name: 'first-sidebar' },
		'My sidebar',
	);
}

registerPlugin( {
	name: 'demo-sidebar',

	render: MyPlugin,
} );
```

Reply:
Then, when rendering as an element, it will render, but with warnings, because we're passing props (why are we passing props?):

```js
const { createElement } = wp.element;
const { registerPlugin, PluginSidebar } = wp.editPost.__experimental;

function MyPlugin() {
	return createElement(
		PluginSidebar,
		{ name: 'first-sidebar' },
		createElement( 'p', null, 'My sidebar' )
	);
}

registerPlugin( {
	name: 'demo-sidebar',

	render: MyPlugin,
} );
```